src/overlays/actors/ovl_Bg_Mori_Hineri/z_bg_mori_hineri.c

```diff
@@ -20,23 +35,251 @@ const ActorInit Bg_Mori_Hineri_InitVars = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hineri/func_808A3BF0.s")
+        unk_14E = globalCtx->cameraPtrs[0]->unk_14E;
+        if ((unk_14E != 0) && (globalCtx->cameraPtrs[unk_14E]->setting == 0x26)) {
+            func_800803F0(globalCtx, unk_14E);
+        }
+        func_800800F8(globalCtx, 0xCBC, 0x28, &this->dyna.actor, 0);
+        D_808A43E0 = func_800800F8(globalCtx, 0xCBD, 0x28, &this->dyna.actor, 0);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hineri/func_808A3BFC.s")
+void func_808A3E54(BgMoriHineri* this, GlobalContext* globalCtx) {
+    s8 objBankIndex;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hineri/func_808A3C8C.s")
+    if (globalCtx->activeCamera == D_808A43E0) {
+        if (D_808A43E0 != 0) {
+            objBankIndex = this->dyna.actor.objBankIndex;
+            this->dyna.actor.objBankIndex = this->objBankIndex;
+            this->objBankIndex = objBankIndex;
+            this->dyna.actor.params ^= 1;
+            D_808A43E0 = 0;
+            func_80078884(NA_SE_SY_TRE_BOX_APPEAR);
+        } else {
+            this->dyna.actor.draw = NULL;
+            this->actionFunc = func_808A3D58;
+            D_808A43E0 = -1;
+        }
+    }
+    if ((D_808A43E0 > 0) &&
+        (((ACTIVE_CAM->eye.z - this->dyna.actor.posRot.pos.z) < 1100.0f))) {
+        func_8002F948(&this->dyna.actor, NA_SE_EV_FLOOR_ROLLING - SFX_FLAG);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hineri/func_808A3D58.s")
+void BgMoriHineri_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgMoriHineri* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hineri/func_808A3E54.s")
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hineri/BgMoriHineri_Update.s")
+void func_808A3F58(Actor* thisx, GlobalContext* globalCtx) {
+    
+    BgMoriHineri* this = THIS;
+    s8 objIndex;
+    MtxF mtx;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    Gfx* dispRefs[4];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Hineri/func_808A3F58.s")
+    Graph_OpenDisps(&dispRefs, globalCtx->state.gfxCtx, "../z_bg_mori_hineri.c", 611);
+    func_80093D18(globalCtx->state.gfxCtx);
+    gSPSegment(gfxCtx->polyOpa.p++, 0x08, globalCtx->objectCtx.status[this->moriTexObjIdx].segment);
+    gSPMatrix(gfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_mori_hineri.c", 618),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(gfxCtx->polyOpa.p++, sDLists[this->dyna.actor.params]);
+    if (this->boxObjIdx > 0) {
+        Matrix_Get(&mtx);
+    }
+    if ((this->actionFunc == func_808A3C8C) && (this->dyna.actor.shape.rot.z != 0)) {
+        func_80093D18(globalCtx->state.gfxCtx);
+        if (this->dyna.actor.params == 0) {
+            Matrix_Translate(-1761.0f, 1278.0f, -1821.0f, 0);
```

Comment:
MatrixMode enum values should be used when calling Matrix_ functions with a mode argument, like this one.
```suggestion
            Matrix_Translate(-1761.0f, 1278.0f, -1821.0f, MTXMODE_NEW);
```